composer.json

```diff
@@ -43,13 +43,13 @@
         "ext-sockets": "*",
         "appwrite/php-clamav": "2.0.*",
         "appwrite/php-runtimes": "0.11.*",
-        "utopia-php/abuse": "0.27.*",
+        "utopia-php/abuse": "0.30.*",
         "utopia-php/analytics": "0.10.*",
-        "utopia-php/audit": "0.29.*",
+        "utopia-php/audit": "0.32.*",
         "utopia-php/cache": "0.8.*",
         "utopia-php/cli": "0.15.*",
         "utopia-php/config": "0.2.*",
-        "utopia-php/database": "0.38.*",
+        "utopia-php/database": "0.41.*",
```

Comment:
I have 3 collections with a two-way one-to-one relationship between them like level1 => level2 => level3 and I experienced 2 problems:

## First problem

With update access to level1 but read-only access to level2, I can update a document in level1 to point to a document in level2, but when I switch it to point to another document:

```json
{
  "level2": "level2other"
}
```
I get an error:

```json
{
    "message": "Document with the requested ID already exists.",
    "code": 409,
    "type": "document_already_exists",
    "version": "dev",
    "file": "/usr/src/code/app/controllers/api/databases.php",
    "line": 3286,
    "trace": [
        {
            "file": "/usr/src/code/vendor/utopia-php/framework/src/App.php",
            "line": 593,
            "function": "{closure}",
            "args": [
                "one-to-one",
                "level1",
                "level1a",
                {
                    "$internalId": "1",
                    "$id": "level1a",
                    "$createdAt": "2023-08-10T23:15:31.935+00:00",
                    "$updatedAt": "2023-08-11T00:19:12.395+00:00",
                    "$permissions": [],
                    "$collection": "level1",
                    "level2": "level2a"
                },
                [],
                null,
                {},
                {},
                {},
                "default"
            ]
        },
        {
            "file": "/usr/src/code/vendor/utopia-php/framework/src/App.php",
            "line": 770,
            "function": "execute",
            "class": "Utopia\\App",
            "type": "->",
            "args": [
                {},
                {}
            ]
        },
        {
            "file": "/usr/src/code/app/http.php",
            "line": 248,
            "function": "run",
            "class": "Utopia\\App",
            "type": "->",
            "args": [
                {},
                {}
            ]
        }
    ]
}
```

## Second problem

With update access to level1 and level2 but read-only access to level3, I can't set level3 on level2:

```json
{
  "level2": {
    "$id": "level2b",
    "level3": "level3a"
  }
}
```

results in:

```
{
    "$id": "level1a",
    "$createdAt": "2023-08-10T23:15:31.935+00:00",
    "$updatedAt": "2023-08-11T00:25:44.052+00:00",
    "$permissions": [],
    "level2": {
        "_createdAt": null,
        "$id": "level2b",
        "$updatedAt": "2023-08-11T00:16:59.164+00:00",
        "$permissions": [],
        "level3": null,
        "$databaseId": "one-to-one",
        "$collectionId": "level2"
    },
    "$databaseId": "one-to-one",
    "$collectionId": "level1"
}
```

Reply:
Also, when I sent a request like:

```json
{
  "level2": {
    "$id": "level2b",
    "level3": "level3a"
  }
}
```

my level2 document ended up with a `_createdAt` attribute:

```json
{
  "level2": {
    "_createdAt": null,
    "$id": "level2a",
    "$updatedAt": "2023-08-10T23:15:53.689+00:00",
    "$permissions": [],
    "level3": null,
    "$databaseId": "one-to-one",
    "$collectionId": "level2"
  }
}
```